src/ds/defines.h

```diff
@@ -62,10 +62,15 @@ namespace snmalloc
 #else
 #  if __has_builtin(__builtin_assume)
 #    define SNMALLOC_ASSUME(x) __builtin_assume((x))
-#  else
+#  elif defined(_MSC_VER)
+#    define SNMALLOC_ASSUME(x) __assume((x));
+#  elif defined(__GNUC__)
 #    define SNMALLOC_ASSUME(x) \
-      do \
-      { \
-      } while (0)
+      if (!(x)) \
+        __builtin_unreachable();
+#  else
+do
+{
+} while (0)
```

Comment:
This is missing the #define and backlashes.

Does this path even get used? `__GNUC__ˋ includes both GCC and clang.

Reply:
Should have realised, was very confused by clangformat on that one.

Reply:
Not used by anything yet.